operator/pkg/apis/istio/v1alpha1/validation/validation.go

```diff
@@ -160,12 +165,43 @@ func checkDeprecatedSettings(iop *v1alpha1.IstioOperatorSpec) (util.Errors, stri
 			}
 		}
 	}
-	return errs, strings.Join(messages, "\n")
+	return errs, messages
 }
 
 // validateFeatures check whether the config sematically make sense. For example, feature X and feature Y can't be enabled together.
-func validateFeatures(values *valuesv1alpha1.Values, spec *v1alpha1.IstioOperatorSpec) util.Errors {
-	return CheckServicePorts(values, spec)
+func validateFeatures(values *valuesv1alpha1.Values, spec *v1alpha1.IstioOperatorSpec) (util.Errors, []string) {
+	errs := CheckServicePorts(values, spec)
+	warningMessages := checkAutoScaleAndReplicaCount(values, spec)
+	return errs, warningMessages
+}
+
+// checkAutoScaleAndReplicaCount warns when autoscaleEnabled is true and k8s replicaCount is set.
+func checkAutoScaleAndReplicaCount(values *valuesv1alpha1.Values, spec *v1alpha1.IstioOperatorSpec) []string {
```

Comment:
can this check find not only warning but also errors? For example, the replicaCount set but also autoscale enabled, should that be considered an error? I wonder if it is actually better to return both errors and warning by this method? which will bear a similar interface as method validateFeatures? Since you are changing the signature of validateFeatures, do you mind getting this new method same interface?

Reply:
> can this check find not only warning but also errors? For example, the replicaCount set but also autoscale enabled, should that be considered an error? I wonder if it is actually better to return both errors and warning by this method? which will bear a similar interface as method validateFeatures? Since you are changing the signature of validateFeatures, do you mind getting this new method same interface?

@litong01 I think warnings should be better, because in general the change in replica count doesn't hurt too much, and it only occurred when you enable and then disable the autoscalEanbled. However using errors will just break the current IstioOperator CR, which is a compatibility issue.

Reply:
@timonwong OK. Thanks. then I am ok with the changes. Thanks.